### PR TITLE
Release/0.1.1

### LINKED ISSUE #1: 1) Bug fix: Sending messages
There are some bugs in our starting code that we would like you to investigate and resolve. First, when you send a new message, it does not immediately appear on the screen. We would like new messages to be immediately added to the chat UI for both existing conversations and new conversations. (Note: This may require more than one change)
Additionally, messages are not displayed in the correct order on page load. Messages should be displayed in order with the oldest messages at the top and newest messages at the bottom.

### PR DESCRIPTION
**Pull request by dillonupgradeit**
Fixes #1 
1. First, when you send a new message, it does not immediately appear on the screen
  + Fixed state update in reducerFunctions.addMessageToStore
  + Fixed state update in reducerFunctions.addNewConvoToStore
  + Added async to thunkCreators.postMessage
2. Additionally, messages are not displayed in the correct order on page load.
  + Fixed message sort order in routes.api.conversations
  + Fixed conversation.lastMessage in routes.api.conversations
  + Fixed conversations sort order in routes.api.conversations

### CODE REVIEW

File: client/src/store/utils/reducerFunctions.js [hunk 1]
```diff
@@ -13,9 +13,10 @@ export const addMessageToStore = (state, payload) => {
 
   return state.map((convo) => {
     if (convo.id === message.conversationId) {
-      convo.messages.push(message);
-      convo.latestMessageText = message.text;
-      return convo;
+      const convoCopy = { ...convo };
+      convoCopy.messages.push(message);
+      convoCopy.latestMessageText = message.text;
+      return convoCopy;
```
Comment (FilipeGorodscy) [inline client/src/store/utils/reducerFunctions.js:19]: 💯

File: client/src/store/utils/reducerFunctions.js [hunk 2]
```diff
@@ -69,10 +70,11 @@ export const addSearchedUsersToStore = (state, users) => {
...
     if (convo.otherUser.id === recipientId) {
-      convo.id = message.conversationId;
-      convo.messages.push(message);
-      convo.latestMessageText = message.text;
-      return convo;
+      const convoCopy = { ...convo };
+      convoCopy.id = message.conversationId;
+      convoCopy.messages.push(message);
+      convoCopy.latestMessageText = message.text;
+      return convoCopy;
```
Comment (FilipeGorodscy) [inline client/src/store/utils/reducerFunctions.js:77]: 💯

File: client/src/store/utils/thunkCreators.js [hunk 3]
```diff
@@ -93,9 +93,9 @@ const sendMessage = (data, body) => {
 
 // message format to send: {recipientId, text, conversationId}
 // conversationId will be set to null if its a brand new conversation
-export const postMessage = (body) => (dispatch) => {
+export const postMessage = (body) => async (dispatch) => {
   try {
-    const data = saveMessage(body);
+    const data = await saveMessage(body);
```
Comment (FilipeGorodscy) [inline client/src/store/utils/thunkCreators.js:98]: 💯

File: server/routes/api/conversations.js [hunk 4]
```diff
@@ -19,9 +19,9 @@ router.get("/", async (req, res, next) => {
         },
       },
       attributes: ["id"],
-      order: [[Message, "createdAt", "DESC"]],
+      order: [[Message, "createdAt", "ASC"]],
       include: [
-        { model: Message, order: ["createdAt", "DESC"] },
+        { model: Message },
```
Comment (FilipeGorodscy) [inline server/routes/api/conversations.js:24]: Awesome!

File: server/routes/api/conversations.js [hunk 5]
```diff
@@ -68,10 +68,15 @@ router.get("/", async (req, res, next) => {
...
-      convoJSON.latestMessageText = convoJSON.messages[0].text;
+      convoJSON.latestMessageText = convoJSON.messages[convoJSON.messages.length-1].text;
       conversations[i] = convoJSON;
     }
 
+    // sort conversations based on most recent message createdAt date
+    conversations.sort((convoA,convoB) => {
+      return convoB.messages[convoB.messages.length-1].createdAt - convoA.messages[convoA.messages.length-1].createdAt;
+    });
+
```
Comment (FilipeGorodscy) [inline server/routes/api/conversations.js:79]: Perfect! 💯

**Review by FilipeGorodscy — COMMENTED** [review]
Awesome work on this PR! I'll let another reviewer take a look and give you the approval to merge! 💯

**Review by Tahoum — APPROVED** [review]
Looks good to me

**Pull request merged**